sdcm/utils/diagnostic_collector/diagnostic_manager.py

```
@@ -0,0 +1,165 @@
+
+    def get_results(self) -> list[DiagnosticResult]:
+        """Get the collected diagnostics results."""
+        with self._results_lock:
+            return self._results.copy()
+
+    def clear_results(self):
+        """Clear the collected diagnostics results."""
+        with self._results_lock:
+            self._results.clear()
+
+    def __enter__(self) -> DiagnosticManager:
```

Ruff (0.15.18)

[error] Undefined name `DiagnosticManager`

(F821)